Add RadioGroupControl support for "oneOf" enums

AlexandraBuzila opened the pull request:
- the radio group control can now handle 'oneOf' schemas as well. If the
oneOf item has a title, it will be shown as the label for the option
- update Radio Group example with a oneOf demo
- add test

sdirix reviewed (changes requested):
Thanks for the PR!

I would prefer when we could use the same approach as with the regular enum renderers, i.e. use `withJsonFormsEnumProps` for the `MaterialRadioGroupControl` and `withJsonFormsOneOfEnumProps` for the `MaterialOneOfRadioGroupControl`, both of them using the same control.

AlexandraBuzila commented:
I rebased the branch and added a dedicated control `MaterialOneOfRadioGroupControl` for the oneof enums. The code looks a lot cleaner now, thank you for the review!

sdirix reviewed (approved):
LGTM
